### PR TITLE
Adding asciidocfx version 1.5.5

### LINKED ISSUE #33620: Cask request: AsciidocFX
### Cask details

Name: AsciidocFX

Homepage: https://github.com/asciidocfx/AsciidocFX

Download URL: https://github.com/asciidocfx/AsciidocFX/releases/latest

Description: AsciidocFX is a book/document editor to build PDF, Epub, Mobi and HTML books, documents and slides.

### PR DESCRIPTION
**Pull request by gobinathm**
- New cask for asciidocfx

---
Closes #33620

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] `sha256` was updated, but `version` remained the same.
      [Post a link to a public confirmation by the developer.][version-checksum]:

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by commitay** [comment]
@gobinathm This cask has a `JavaApplicationStub` `installer`, does it work with an install script like [this](
https://github.com/caskroom/homebrew-cask/blob/008c4c512d9ab6ae167e9dce81960ad912978e9d/Casks/owasp-zap.rb#L13-L17)?

**Comment by gobinathm** [comment]
@commitay valid point let me check

File: Casks/asciidocfx.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+cask 'asciidocfx' do
+  version '1.5.5'
+  sha256 '368756a759175ff5e0a321db8b5665928bcc5b18eadeba4a788ce76ff6ab3c75'
+
+  # github.com/asciidocfx/AsciidocFX was verified as official when first introduced to the cask
+  url "https://github.com/asciidocfx/AsciidocFX/releases/download/v#{version}/AsciidocFX_Mac.dmg"
```
Comment (reitermarkus) [inline Casks/asciidocfx.rb]: This needs an [`appcast`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md).

Comment (miccal) [inline Casks/asciidocfx.rb]: Done.

**Comment by commitay** [comment]
This still needed a correct install and uninstall.

**Pull request merged**